src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp

```diff
@@ -95,7 +904,1269 @@ class boss_thorim : public CreatureScript
+                    _events.ScheduleEvent(EVENT_THIRD_ABILITY, urand(6000, 8000));
+                if (_info->Type == DARK_RUNE_CHAMPION)
+                    _events.ScheduleEvent(EVENT_ABILITY_CHARGE, 8000);
+            }
+
+            void EnterCombat(Unit* /*who*/) override
+            {
+                if (_info->Type == DARK_RUNE_WARBRINGER)
+                    DoCast(me, SPELL_AURA_OF_CELERITY);
+
+                if (!_isInArena)
+                    if (Creature* colossus = _instance->GetCreature(DATA_RUNIC_COLOSSUS))
+                        colossus->AI()->DoAction(ACTION_ACTIVATE_RUNIC_SMASH);
+            }
+
+            void EnterEvadeMode(EvadeReason why) override
+            {
+                if (why != EVADE_REASON_NO_HOSTILES && why != EVADE_REASON_BOUNDARY)
+                    return;
+
+                // this should only happen if theres no alive player in the arena -> summon orb
+                if (Creature* thorim = _instance->GetCreature(BOSS_THORIM))
+                    thorim->AI()->DoAction(ACTION_BERSERK);
+                ScriptedAI::EnterEvadeMode(why);
+            }
+
+            void ExecuteEvent(uint32 eventId) override
+            {
+                switch (eventId)
+                {
+                    case EVENT_PRIMARY_ABILITY:
+                        if (UseAbility(_info->PrimaryAbility))
+                            _events.Repeat(3000, 6000);
+                        else
+                            _events.Repeat(1000);
+                        break;
+                    case EVENT_SECONDARY_ABILITY:
+                        if (UseAbility(_info->SecondaryAbility))
+                            _events.Repeat(12000, 16000);
+                        else
+                            _events.Repeat(1000);
+                        break;
+                    case EVENT_THIRD_ABILITY:
+                        if (UseAbility(_info->ThirdAbility))
+                            _events.Repeat(6000, 8000);
+                        else
+                            _events.Repeat(1000);
+                        break;
+                    case EVENT_ABILITY_CHARGE:
+                    {
+                        Unit* referer = me;
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, [referer](Unit* unit){ return unit->GetTypeId() == TYPEID_PLAYER && unit->IsInRange(referer, 8.0f, 25.0f); }))
+                            DoCast(target, SPELL_CHARGE);
+                        _events.ScheduleEvent(eventId, 12000);
+                        break;
+                    }
+                    default:
+                        break;
+                }
+            }
+
+        private:
+            bool _isInArena;
+        };
+
+        CreatureAI* GetAI(Creature* creature) const override
+        {
+            return GetUlduarAI<npc_thorim_arena_phaseAI>(creature);
+        }
+};
+
+struct npc_thorim_minibossAI : public ScriptedAI
```

Comment:
this script has no conditions for the unit to not attack arena players (might get aggro from healing or similar)